src/Resolver/Proxy.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Upward\Resolver;
+
+use Magento\Upward\Definition;
+use Zend\Http\Client;
+
+class Proxy extends AbstractResolver
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getIndicator(): string
+    {
+        return 'proxy';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isValid(Definition $definition): bool
+    {
+        $rootDefinition = $this->getIterator()->getRootDefinition();
+        if ($rootDefinition->has('proxy')) {
+            $proxyDefinition = $rootDefinition->get('proxy');
+            if (!$proxyDefinition->has('target')) {
+                return false;
+            }
+        } else {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function resolve($definition)
+    {
+        $target          = $this->getIterator()->get('proxy.target');
```

Comment:
Again, should be working off of the passed `$definition` not looking at some other key in the root.